File: core/src/main/resources/scripts/kestra.js
function Kestra() {

}

Kestra._send = (map) => {
    console.log("::" + JSON.stringify(map) + "::");
}

Kestra._metrics = (name, type, value, tags) => {
    Kestra._send({
        "metrics": [
            {
                "name": name,
                "type": type,
                "value": value,
                tags: tags || {}
            }
        ]
    });
}

Kestra.outputs = (outputs) => {
    Kestra._send({
        "outputs": outputs
    });
}

Kestra.counter = (name, value, tags) => {
    Kestra._metrics(name, "counter", value, tags);
}

Kestra.timer = (name, duration, tags) => {
    if (typeof duration === "function") {
        const start = new Date();
        duration(() => {
            Kestra._metrics(name, "timer", (new Date() - start) / 1000, tags);
        });
    } else {
        Kestra._metrics(name, "timer", duration, tags)
    }
}

module.exports = Kestra;

File: core/src/main/resources/scripts/python/src/kestra.py
import json
from datetime import datetime

class Kestra:
    def __init__(self):
        pass

    @staticmethod
    def _send(map):
        print("::" + json.dumps(map) + "::")

    @staticmethod
    def _metrics(name, type, value, tags=None):
        Kestra._send({
            "metrics": [
                {
                    "name": name,
                    "type": type,
                    "value": value,
                    "tags": tags or {}
                }
            ]
        })

    @staticmethod
    def outputs(map):
        Kestra._send({
            "outputs": map
        })

    @staticmethod
    def counter(name, value, tags=None):
        Kestra._metrics(name, "counter", value, tags)

    @staticmethod
    def timer(name, duration, tags=None):
        if callable(duration):
            start = datetime.now()
            duration()
            Kestra._metrics(name, "timer", (datetime.now().microsecond - start.microsecond) / 1000, tags)
        else:
            Kestra._metrics(name, "timer", duration, tags);
